src/wp-includes/script-loader.php

```diff
@@ -3671,3 +3671,32 @@ function wp_enqueue_classic_theme_styles() {
 		wp_enqueue_style( 'classic-theme-styles' );
 	}
 }
+
+/**
+ * Loads classic theme styles on classic themes in the editor.
+ *
+ * This is needed for backwards compatibility for button blocks specifically.
+ *
+ * @since 6.1.0
+ *
+ * @param array $editor_settings The array of editor settings.
+ * @return array A filtered array of editor settings.
+ */
+function wp_add_editor_classic_theme_styles( $editor_settings ) {
+	if ( wp_is_block_theme() ) {
+		return $editor_settings;
+	}
+	$suffix = wp_scripts_get_suffix();
+	$classic_theme_styles = __DIR__ . "/css/dist/block-library/classic$suffix.css";
+
+	// This follows the pattern of get_block_editor_theme_styles,
+	// but we can't use get_block_editor_theme_styles directly as it
+	// only handles external files or theme files.
+	$editor_settings['styles'][] = array(
+		'css'            => file_get_contents( $classic_theme_styles ),
+		'__unstableType' => 'theme',
```

Comment:
Maybe this should be `core`? It's arguably Core that's providing this styling 🤔

Reply:
I'll tentatively change it to `core` since IMO it better represents where the styling is coming from. 

Since this is targeting classic themes, it shouldn't have any actual implications, since this is for Global Styles -- which are unavailable for Classic Themes.

Reply:
Thanks, that's a good idea.

Reply:
:wave: Looked at where `__unstableType` is used. It's only at https://github.com/WordPress/gutenberg/blob/88a1f7ae18bea3a13f6d1e507e3c25090b76cb08/packages/edit-post/src/editor.js#L165 

If I'm reading the original [PR correctly](https://github.com/WordPress/gutenberg/pull/35736#discussion_r733460913), this is used to decide whether to render a particular stylesheet when the user deactivates "theme styles". If the stylesheet needs to be disabled by that check, we should use "theme", otherwise "core" is fine. See check in this screen:

![imagem](https://user-images.githubusercontent.com/583546/196939214-333df695-74b2-43a2-b25a-ab031099fae1.png)

Reply:
These styles are the "default" ones set for buttons by core, they aren't connected to a specific theme, so I think `core` is correct.